### PR TITLE
Update the jint options to support System.text.json

### PR DESCRIPTION
**Pull request by hyzx86**
try to reproduce `System.InvalidOperationException:“The node must be of type 'JsonObject'.”`

image

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
Your test case here seems to pass. If the issue related to your code?

**Comment by hyzx86** [comment]
Strange. Can't reproduce in OC, I haven't found any reason yet, 
Now I change JObject to an object in my project, and the problem is solved

**Comment by hyzx86** [comment]
The problem recurred. My previous tests were broken, I updated them, and now the problem recurred.


```c#
            var findUser = new GlobalMethod
            {
                Name = "getUserByName",
                Method = sp => (string userName) =>
                {
                    var userManager = sp.GetRequiredService<UserManager<IUser>>();
                    var userInfo = userManager.FindByNameAsync(userName).GetAwaiter().GetResult();
                    var jobjUser = JObject.FromObject(userInfo, JOptions.CamelCase);
                    jobjUser.Remove("securityStamp");
                    jobjUser.Remove("passwordHash");
                    jobjUser.Remove("resetToken");
                    jobjUser.Remove("userTokens");
                    return jobjUser;
                }
            };

            var scriptingEngine = scope.ServiceProvider.GetRequiredService<IScriptingEngine>();
            var scriptingScope = scriptingEngine.CreateScope([findUser], scope.ServiceProvider, null, null);
            var result = (bool)scriptingEngine.Evaluate(scriptingScope, "var user = getUserByName('admin'); return user.userName == 'admin'");
            Assert.True(result);
```

**Comment by hyzx86** [comment]
https://github.com/sebastienros/jint/blob/107f5433293e60dfc96f0bff83c427c6d7081537/Jint.Tests.PublicInterface/InteropTests.SystemTextJson.cs#L32-L59

**Comment by hyzx86** [comment]
will fix by https://github.com/OrchardCMS/OrchardCore/pull/15449

**Comment by hyzx86** [comment]
Hi @MikeAlhayek   ， this PR is ready to merge

File: test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                    ""Email""
+                ],
+                ""Optout"": []
+            },
+            ""UserTimeZone"": {
+                ""TimeZoneId"": null
+            }
+        }";
+
+            var userManager = scope.ServiceProvider.GetRequiredService<UserManager<IUser>>();
```
Comment (MikeAlhayek) [inline test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs:83]: Why do you need to fetch a user and then remove properties? During arrange, manually create a user with some key values to test instead.

File: test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+using OrchardCore.Scripting;
+using OrchardCore.Tests.Apis.Context;
+using OrchardCore.Users;
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Tests.Scripting;
+public class ScriptFunctionsTest
+{
+    [Fact]
+    public async Task ProcessUserInfoTest()
```
Comment (MikeAlhayek) [inline test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs:12]: The name does not example what you are testing or the expected result

File: src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs [hunk 3]
```diff
@@ -19,7 +22,34 @@ public JavaScriptEngine(IMemoryCache memoryCache)
...
+                {
+                    var wrapped = new ObjectWrapper(e, target);
+                    if (target is JsonArray)
+                    {
+                        wrapped.Prototype = e.Intrinsics.Array.PrototypeObject;
+                    }
+                    return wrapped;
+                };
+
+                // we cannot access this[string] with anything else than JsonObject, otherwise itw will throw
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs:38]: When you add comments, please start with a Capital letter and add a period at the end of the sentence.

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

**Comment by hyzx86** [comment]
@MikeAlhayek  Done.

File: test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        using var context = new SiteContext();
+        await context.InitializeAsync();
+        await context.UsingTenantScopeAsync(scope =>
+        {
+            var findUser = new GlobalMethod
+            {
+                Name = "tryAccessJsonObject",
+                Method = sp => () =>
+                {
+                    return new JsonObject
```
Comment (sebastienros) [inline test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs:20]: Could you try adding other STJ types in the test, like arrays, bool, numbers, to be sure it works fine too.

Comment (hyzx86) [reply test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs:20]: the jint project contains the full test, do I need to copy it to oc?

https://github.com/sebastienros/jint/blob/107f5433293e60dfc96f0bff83c427c6d7081537/Jint.Tests.PublicInterface/InteropTests.SystemTextJson.cs#L32-L59

Comment (hyzx86) [reply test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs:20]: The problem is that false value in JsonObject does not work as expected 

Hello, thanks for the PR, but I find that false for Boolean values doesn't seem to handle as expected

Comment (hyzx86) [reply test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs]: relate https://github.com/sebastienros/jint/issues/1801

**Review by sebastienros — APPROVED** [review]

**Pull request merged**